### DISCUSSION TITLE
need help with models fields with relationships to other models.

### DISCUSSION THREAD

**Discussion by chinoosniel** [django-commons/django-unicorn#144] [Q&A]
I ask for help to clarify what I am doing wrong.

```
class Traza(models.Model):
    fecha = models.DateTimeField(default=timezone.now)
    usuario = models.CharField(max_length=100, blank=True, null=True)

class Book(models.Model):
    nombre = models.CharField(max_length=20, null=True)
    user = models.OneToOneField(Traza, .... )


 class ProductorView(UnicornView):
    book: Book = None
    base = 0

  def update_interface(self):
      self.book = Book.objects.all()[self.base]
      self.base += 1
```

If the model "Book" has a field with relationships with another model then it gives the following error:
File "/venv/lib/python3.8/site-packages/django_unicorn/views.py", line 116, in _set_property_from_data
    _set_property_from_data(field, field.name, value)
AttributeError: 'Traza' object has no attribute 'name'

However, if the relationship is empty, it does not happen. This only happens when a field is related to another model and has an assigned value.

Thanks in advance

**Comment by adamghill** [comment]
I created https://github.com/adamghill/django-unicorn/issues/145 because I think it's a bug, so I'm going to lock this thread.

Thanks for letting me know! I'm going to fix it as soon as I can.

**Comment by adamghill** [comment]
I think the new version 0.26.0 should fix this issue. I changed a lot of code behind-the-scenes to handle Django models and QuerySets better including ForeignKeys. Please check it out and let me know if you run into any issues!

**Reply by adamghill** [reply]
0.27.2 has more fixes for this, as well. Hopefully that version works correctly!
